src/backend/distributed/executor/multi_utility.c

```diff
@@ -336,6 +337,25 @@ multi_ProcessUtility(PlannedStmt *pstmt,
 		ErrorIfUnsupportedTruncateStmt((TruncateStmt *) parsetree);
 	}
 
+	if (IsA(parsetree, DropStmt))
+	{
+		DropStmt *dropStatement = (DropStmt *) parsetree;
+		bool dropCitus = IsCitusDropExtensionStmt(dropStatement);
+
+		/*
+		 * We need to take a lock on pg_dist_node while dropping the
+		 * citus extension. Otherwise, deadlock between dropping
+		 * extension and deadlock detection daemon may occur. We take
+		 * the lock here because standard process utility does not take
```

Comment:
I'd mention a bit more explicitly that `pg_dist_node` is already going to be dropped before starting the sentence `We take ...`